conans/client/cmake.py

```diff
@@ -117,14 +138,16 @@ def _cmake_cross_build_defines(self, the_os, os_ver):
         ret = OrderedDict()
 
         # SYSTEM NAME
-        env_system_name = get_env("CONAN_CMAKE_SYSTEM_NAME", "")
         os_ver = get_env("CONAN_CMAKE_SYSTEM_VERSION", os_ver)
-        if env_system_name and env_system_name != "False":  # False means not auto-set
-            ret["CMAKE_SYSTEM_NAME"] = env_system_name
+        # String not empty
+        if not isinstance(self._cmake_system_name, bool) and self._cmake_system_name != "":
```

Comment:
Simplify to ``if self._cmake_system_name``. As it is never True. Maybe the issue is that this flag model is weird